das/factory.go

```diff
@@ -148,3 +200,218 @@ func CreateBatchPosterDAS(
+				return nil, nil, nil, errors.New("l1-node-url and sequencer-inbox-address must be specified along with sync-to-storage.eager")
+			}
+			storageService, err = NewSyncingFallbackStorageService(
+				ctx,
+				storageService,
+				restAgg,
+				l1Reader,
+				*seqInboxAddress,
+				syncConf)
+			dasLifecycleManager.Register(storageService)
+			if err != nil {
+				return nil, nil, nil, err
+			}
+		} else {
+			storageService = NewFallbackStorageService(storageService, restAgg,
+				retentionPeriodSeconds, syncConf.IgnoreWriteErrors, true)
+			dasLifecycleManager.Register(storageService)
+		}
+
+	}
+
+	var daWriter DataAvailabilityServiceWriter
+	var daReader DataAvailabilityServiceReader = storageService
+
+	if config.KeyConfig.KeyDir != "" || config.KeyConfig.PrivKey != "" {
+		var seqInboxCaller *bridgegen.SequencerInboxCaller
+		if seqInboxAddress != nil {
+			seqInbox, err := bridgegen.NewSequencerInbox(*seqInboxAddress, (*l1Reader).Client())
+			if err != nil {
+				return nil, nil, nil, err
+			}
+
+			seqInboxCaller = &seqInbox.SequencerInboxCaller
+		}
+		if config.DisableSignatureChecking {
+			seqInboxCaller = nil
+		}
+
+		privKey, err := config.KeyConfig.BLSPrivKey()
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		daWriter, err = NewSignAfterStoreDASWriterWithSeqInboxCaller(
+			privKey,
+			seqInboxCaller,
+			storageService,
+			config.ExtraSignatureCheckingPublicKey,
+		)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+	}
+
+	if config.RegularSyncStorageConfig.Enable && len(syncFromStorageServices) != 0 && len(syncToStorageServices) != 0 {
+		regularlySyncStorage := NewRegularlySyncStorage(syncFromStorageServices, syncToStorageServices, config.RegularSyncStorageConfig)
+		regularlySyncStorage.Start(ctx)
+	}
+
+	if seqInboxAddress != nil {
+		daReader, err = NewChainFetchReader(daReader, (*l1Reader).Client(), *seqInboxAddress)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+	}
+
+	return daReader, daWriter, dasLifecycleManager, nil
+}
+
+func CreateDAReaderForNode(
+	ctx context.Context,
+	config *DataAvailabilityConfig,
+	l1Reader *headerreader.HeaderReader,
+	seqInboxAddress *common.Address,
+) (DataAvailabilityServiceReader, *LifecycleManager, error) {
+	if !config.Enable {
+		return nil, nil, nil
+	}
+
+	// Check config requirements
+	if config.AggregatorConfig.Enable {
+		return nil, nil, errors.New("node.data-availability.rpc-aggregator is only for Batch Poster mode")
+	}
+	if config.KeyConfig.KeyDir != "" || config.KeyConfig.PrivKey != "" {
+		return nil, nil, errors.New("node.data-availability.key options are only for daserver committee members")
+	}
+
+	if config.LocalDBStorageConfig.Enable || config.LocalFileStorageConfig.Enable || config.S3StorageServiceConfig.Enable {
+		return nil, nil, errors.New("None of --node.data-availability.(local-db-storage|local-file-storage|s3-storage).enable may be set when running a Nitro AnyTrust node in non-Batch Poster mode.")
+	}
```

Comment:
Looks like we also block this in batch poster mode?

Reply:
I don't think it's worth the complexity, but theoretically we could drop these options from the node if they're not valid in any config (daserver only)

Reply:
Thanks for pointing this out. I made it so there are now two separate functions for setting up the config, one for nitro nodes and one for daserver, that only add the config that they use. This means all this kind of validation can be removed.

Reply:
Oooh amazing. I was thinking about if that'd be possible, but I thought it would be too much work